File: app/src/main/java/com/example/visiba/visibarectest/Adapters/ImageAdapter.java
package com.example.visiba.visibarectest.Adapters;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;

import com.example.visiba.visibarectest.AppImage;

import java.util.ArrayList;

public class ImageAdapter extends BaseAdapter {
    private Context mContext;
    private ArrayList<AppImage> images;

    public ImageAdapter(Context c, ArrayList<AppImage> images) {
        mContext = c;
        this.images = images;
    }

    public int getCount() {
        return images.size();
    }

    public Object getItem(int position) {
        return images.get(position);
    }

    public long getItemId(int position) {
        return 0;
    }

    public View getView(int position, View convertView, ViewGroup parent) {
        ImageView imageView;
        if (convertView == null) {
            imageView = new ImageView(mContext);
        } else {
            imageView = (ImageView) convertView;
        }

        imageView.setImageDrawable(images.get(position).drawableImage);
        imageView.setAdjustViewBounds(true);

        return imageView;
    }
}

File: app/src/main/java/com/example/visiba/visibarectest/Adapters/ImageAdapter.kt
package com.example.visiba.visibarectest.Adapters

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import com.example.visiba.visibarectest.AppImage

class ImageAdapter(private val mContext: Context, private val images: MutableList<AppImage>) : BaseAdapter() {
    override fun getCount(): Int {
        return images.size
    }

    override fun getItem(position: Int): Any {
        return images[position]
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val imageView: ImageView
        if (convertView == null) {
            imageView = ImageView(mContext)
        } else {
            imageView = convertView as ImageView
        }

        imageView.setImageDrawable(images[position].getDrawable(mContext))
        imageView.adjustViewBounds = true

        return imageView
    }
}

